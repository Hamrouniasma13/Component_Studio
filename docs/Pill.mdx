---
name: Pill
route: /pill
menu: ⚛️ Atoms
---

import { Playground } from 'docz'
import { Pill } from '../src'

# Pill

<Playground>
  <Pill>Status</Pill>
</Playground>

## Size

Pill can be bigger with `large`

<Playground>
  <Pill large>Status</Pill>
</Playground>

## Color

You can use `color` to change the background color of the pill. The primary color of the theme is used by default, and the text color is automatically calculed.

<Playground>
  <Pill color="secondary">Status</Pill> <Pill color="third">Status</Pill>{' '}
  <Pill color="fourth">Status</Pill> <Pill color="fifth">Status</Pill>{' '}
  <Pill color="sixth">Status</Pill>
</Playground>
