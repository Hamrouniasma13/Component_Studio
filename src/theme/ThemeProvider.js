import React from 'react'
import { default as defaultTheme } from '../theme'
import { ThemeProvider as EmotionThemeProvider } from 'emotion-theming'
import { merge } from 'lodash'

export const ThemeContext = React.createContext(defaultTheme)

// Used in Admin portal, for color selector
export const ThemeContextProvider = ({ children, theme = {} }) => {
  return (
    <ThemeContext.Provider value={merge({}, defaultTheme, theme || {})}>
      {children}
    </ThemeContext.Provider>
  )
}

// Main theme provider, used in all portals and website studio
export default function ThemeProvider({ children, theme }) {
  return (
    <EmotionThemeProvider theme={merge({}, defaultTheme, theme || {})}>
      {children}
    </EmotionThemeProvider>
  )
}
