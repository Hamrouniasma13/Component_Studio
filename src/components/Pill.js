/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import React from 'react'
import PropTypes from 'prop-types'
import { textColor, darken } from '../theme/utils'

export default function Pill({ color, large, css, ...props }) {
  const theme = React.useContext(ThemeContext)
  const background = theme.colors[color] || color
  return (
    <span
      css={{
        label: 'pill',
        backgroundColor: background,
        borderRadius: '4px',
        color: textColor(darken(background, 50), theme),
        fontWeight: 'bold',
        fontSize: large ? '16px' : '14px',
        padding: large ? '4px 14px' : '3px 12px',
        ...css
      }}
      {...props}
    />
  )
}

Pill.propTypes = {
  color: PropTypes.string,
  large: PropTypes.bool
}

Pill.defaultProps = {
  color: 'primary'
}
