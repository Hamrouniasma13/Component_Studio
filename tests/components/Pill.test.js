import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

// create matcher in order to access to component's style
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';
import { mount, render } from 'enzyme';

import ThemeProvider from '../../components/ThemeProvider';
import Pill from '../../components/Pill';

expect.extend(createMatchers(emotion));

describe('Component: <Pill />', () => {
  it('render label', () => {
    const tree = renderer.create(<Pill large>News</Pill>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render primary color', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={{ colors: { primary: 'red' } }}>
          <Pill color="primary" small>
            News
          </Pill>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toHaveStyleRule('background-color', 'red');
  });

  it('trigger onClick() callback', () => {
    const onPillClick = jest.fn();
    const wrapper = mount(
      <Pill color="primary" onClick={onPillClick} className="my-pill">
        News
      </Pill>
    );
    wrapper.find('span.my-pill').simulate('click');
    expect(onPillClick).toBeCalled();
  });
});
