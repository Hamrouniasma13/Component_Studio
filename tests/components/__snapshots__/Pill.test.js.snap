// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Pill /> render label 1`] = `
.emotion-0 {
  background-color: #000;
  border-radius: 4px;
  color: #ffffff;
  display: inline;
  font-size: 16px;
  padding: 4px 14px;
}

<span
  className="emotion-0"
  color="#000"
  onClick={[Function]}
  to={undefined}
>
  News
</span>
`;
